locks/locks.go

```diff
@@ -1,22 +1,49 @@
 package locks
 
-var lockRequests chan *lockRequest
+var lockRequests = make(chan *lockRequest)
 
 func init() {
-	lockRequests = make(chan *lockRequest)
 	go locker()
 }
 
+type Locker interface {
+	Lock() Unlocker
+}
+
 // Unlocker Interface for unlocking a key that was provided to Lock()
 type Unlocker interface {
 	Unlock()
 }
 
+type KeyLockerImpl struct {
```

Comment:
I find it confusing that KeyLockerImpl is declared here but then nop is declared and all its methods follow and then we jump back to keylocker. can you rearrange such that keylockerimpl is near its functions?